### PR TITLE
fix/ include active orders into balance limit calculations and other typing issues

### PR DESCRIPTION
**Pull request by Nullably**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue:
#2177 
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:

Special thanks to @TheHolyRoger for the initial contribution

### CODE REVIEW

File: hummingbot/market/kucoin/kucoin_market.pyx [hunk 1]
```diff
@@ -657,9 +657,6 @@ cdef class KucoinMarket(MarketBase):
     def ready(self) -> bool:
         return all(self.status_dict.values())
 
-    def get_all_balances(self) -> Dict[str, Decimal]:
-        return self._account_balances.copy()
-
```
Comment (CrimsonJacket) [inline hummingbot/market/kucoin/kucoin_market.pyx:662]: Redundant `get_all_balances()` inherits the same method from `MarketBase`

File: hummingbot/market/market_base.pyx [hunk 2]
```diff
@@ -71,6 +71,26 @@ cdef class MarketBase(NetworkIterator):
...
+        asset_balances = {}
+        for order in [o for o in in_flight_orders.values() if not (o.is_done or o.is_failure or o.is_cancelled)]:
+            if order.trade_type is TradeType.BUY:
+                order_value = Decimal(order.amount * order.price)
+                outstanding_value = order_value - order.executed_amount_quote
+                if order.quote_asset not in asset_balances:
+                    asset_balances[order.quote_asset] = s_decimal_0
+                asset_balances[order.quote_asset] += outstanding_value
+            else:
+                outstanding_value = order.amount = order.executed_amount_base
```
Comment (Nullably) [inline hummingbot/market/market_base.pyx:88]: small mistake here, should be "-", not "=".

File: hummingbot/market/market_base.pyx [hunk 3]
```diff
@@ -138,15 +157,15 @@ cdef class MarketBase(NetworkIterator):
...
-
         for asset_name, available_balance in self._account_available_balances.items():
             if asset_name.upper() in exchange_limits:
                 asset_limit = Decimal(exchange_limits[asset_name.upper()])
-                self._account_available_balances.update({asset_name, min(available_balance, asset_limit)})
+
+                active_asset_balance = self.in_flight_asset_balances(self.in_flight_orders).get(asset_name.upper(), s_decimal_0)
+                current_asset_available_balance = self._account_available_balances.get(asset_name.upper(), s_decimal_0)
+
+                new_asset_available_balance = min(Decimal(current_asset_available_balance + active_asset_balance), asset_limit)
```
Comment (Nullably) [inline hummingbot/market/market_base.pyx:167]: this should be 
new_asset_available_balance = min(current_asset_available_balance, max(asset_limit - active_asset_balance, s_decimal_0))

**Comment by TheHolyRoger** [comment]
Thank you @CrimsonJacket :) 👍 

(Apologies for the sassiness, as per usual ;) )

Looks good to me let's get this in and fix dev branch

**Pull request merged**